commit 963a70bee5880640d0fd83ed29dc1e7ec0d2bd4a
Merge: adb67b373a68 90d74fdbd805
Author: Linus Torvalds <torvalds@linux-foundation.org>
Date:   Fri Aug 19 09:46:11 2022 -0700

    Merge tag 'sound-6.0-rc2' of git://git.kernel.org/pub/scm/linux/kernel/git/tiwai/sound
    
    Pull sound fixes from Takashi Iwai:
     "The only significant core change is ASoC DPCM fix for asymmetric
      setup; other remaining changes are device-specific fixes, including
      the hardening of string manipulations.
    
      One change in platform/x86 is the patch I forgot to apply from a
      series for CS35L41 codec"
    
    * tag 'sound-6.0-rc2' of git://git.kernel.org/pub/scm/linux/kernel/git/tiwai/sound: (21 commits)
      ALSA: hda/realtek: Add quirk for Clevo NS50PU, NS70PU
      ALSA: info: Fix llseek return value when using callback
      ALSA: hda/cs8409: Support new Dolphin Variants
      platform/x86: serial-multi-instantiate: Add CLSA0101 Laptop
      ALSA: hda/realtek: Add quirk for Lenovo Yoga7 14IAL7
      ALSA: hda: cs35l41: Clarify support for CSC3551 without _DSD Properties
      ALSA: hda/realtek: Add quirks for ASUS Zenbooks using CS35L41
      ASoC: codec: tlv320aic32x4: fix mono playback via I2S
      ASoC: rt5640: Fix the JD voltage dropping issue
      ASoC: tas2770: Fix handling of mute/unmute
      ASoC: tas2770: Drop conflicting set_bias_level power setting
      ASoC: tas2770: Allow mono streams
      ASoC: tas2770: Set correct FSYNC polarity
      ASoC: Intel: fix sof_es8336 probe
      ASoC: DPCM: Don't pick up BE without substream
      ASoC: SOF: ipc3-topology: Fix clang -Wformat warning
      ASoC: sh: rz-ssi: Improve error handling in rz_ssi_probe() error path
      ASoC: SOF: Intel: hda: Fix potential buffer overflow by snprintf()
      ASoC: SOF: debug: Fix potential buffer overflow by snprintf()
      ASoC: Intel: avs: Fix potential buffer overflow by snprintf()
      ...

diff --cc drivers/platform/x86/serial-multi-instantiate.c
index 67feed25c9db,e98007197cf5..5362f1a7b77c
--- a/drivers/platform/x86/serial-multi-instantiate.c
+++ b/drivers/platform/x86/serial-multi-instantiate.c
@@@ -324,10 -325,11 +324,11 @@@ static const struct smi_node cs35l41_hd
  static const struct acpi_device_id smi_acpi_ids[] = {
  	{ "BSG1160", (unsigned long)&bsg1160_data },
  	{ "BSG2150", (unsigned long)&bsg2150_data },
 -	{ "INT3515", (unsigned long)&int3515_data },
  	{ "CSC3551", (unsigned long)&cs35l41_hda },
 +	{ "INT3515", (unsigned long)&int3515_data },
  	/* Non-conforming _HID for Cirrus Logic already released */
  	{ "CLSA0100", (unsigned long)&cs35l41_hda },
+ 	{ "CLSA0101", (unsigned long)&cs35l41_hda },
  	{ }
  };
  MODULE_DEVICE_TABLE(acpi, smi_acpi_ids);